### PR TITLE
Fix CI checks for IPv6 branch

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
See the title

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS_ARP.c [hunk 1]
```diff
@@ -481,7 +481,8 @@ void vARPRefreshCacheEntry( const MACAddress_t * pxMACAddress,
  * @return Either eARPCacheMiss or eARPCacheHit.
  */
     eARPLookupResult_t eARPGetCacheEntryByMac( MACAddress_t * const pxMACAddress,
-                                               uint32_t * pulIPAddress )
+                                               uint32_t * pulIPAddress,
+                                               struct xNetworkInterface ** ppxInterface )
```
Comment (htibosch) [inline FreeRTOS_ARP.c:485]: Yes, that is a good addition. `ppxInterface` will have to be filled in later.

File: FreeRTOS_IP.c [hunk 2]
```diff
@@ -1187,6 +1187,7 @@ BaseType_t FreeRTOS_NetworkDownFromISR( struct xNetworkInterface * pxNetworkInte
 }
 /*-----------------------------------------------------------*/
 
+/* *INDENT-OFF* */
```
Comment (htibosch) [inline FreeRTOS_IP.c:1190]: Very good, thanks for solving the Uncrustify warning here.

File: test/build-combination/Common/main.c [hunk 3]
```diff
@@ -108,20 +108,17 @@ int main( void )
...
      * but a DHCP server cannot be contacted. */
     FreeRTOS_printf( ( "FreeRTOS_IPInit\n" ) );
-    FreeRTOS_IPInit(
-        ucIPAddress,
-        ucNetMask,
-        ucGatewayAddress,
-        ucDNSServerAddress,
-        ucMACAddress );
+
+    FreeRTOS_IPStart();
```
Comment (htibosch) [inline test/build-combination/Common/main.c:112]: Yes, thank you for this change.

File: test/unit-test/ConfigFiles/FreeRTOSConfig.h [hunk 4]
```diff
@@ -117,7 +117,7 @@ void vConfigureTimerForRunTimeStats( void );    /* Prototype of function that in
  * uses the same semantics as the standard C assert() macro. */
 extern void vAssertCalled( unsigned long ulLine,
                            const char * const pcFileName );
-#define configASSERT( x )    if( x == 0 ) TEST_ABORT()
+#define configASSERT( x )    if( ( x ) == 0 ) TEST_ABORT()
```
Comment (htibosch) [inline test/unit-test/ConfigFiles/FreeRTOSConfig.h:120]: This one saves hundreds of compiler warnings :-)

File: tools/tcp_utilities/plus_tcp_demo_cli.c [hunk 5]
```diff
@@ -945,7 +945,7 @@ static struct freertos_addrinfo * pxDNSLookup( char * pcHost,
 
                 #if ( ipconfigUSE_IPv6 != 0 )
                     {
-                        pxResult->ai_addr = ipPOINTER_CAST( struct freertos_sockaddr *, &( pxResult->xPrivateStorage.sockaddr6 ) );
+                        pxResult->ai_addr = ipPOINTER_CAST( struct freertos_sockaddr *, & ( pxResult->xPrivateStorage.sockaddr6 ) );
```
Comment (htibosch) [inline tools/tcp_utilities/plus_tcp_demo_cli.c:948]: Uh? I have seen this space before:
~~~c
- &( pxResult->xPrivateStorage.sockaddr6 )
+ & ( pxResult->xPrivateStorage.sockaddr6 )
~~~

**Review by htibosch — APPROVED** [review]
Thanks for solving these CI and other warnings,
Hein

**Review by cobusve — APPROVED** [review]

**Pull request merged**